oauth-overview.html.md.erb

```diff
@@ -0,0 +1,106 @@
+---
+title: OAuth Overview 
+owner: API
+---
+
+<strong><%= modified_date %></strong>
+
+## <a id='overview'></a> What is Oauth?
+
+OAuth 2.0 is an open standard for authorization and it provides client applications a 'secure delegated access' to server resources on behalf of a resource owner. OAuth essentially allows access tokens to be issued to third-party clients by an authorization server, with the approval of the resource owner, or end-user. The client uses the access token to access our protected APIs.
+More information you can read at http://oauth.net/2/.
+
+If you are already familiar with OAuth, then all you really need to know about are the two authentication endpoints: 
+
+* authorization endpoint:  https://consent.swisscom.com/c/oauth2/oauth
+* token request endpoint: https://consent.swisscom.com/o/oauth2/token
+
+Before you begin to use an application, you will need a client id, redirect URI, and a client secret key. These details will be used to authenticate your application and verify that the API calls being made are valid.
+
+There are 3 ways to obtain an access token, which can be used instead of a client_id/API key, namely:
+
+* [Authorization Code (3-legged) Tutorial](./oauth-authorisation-code.html)
+* [Client Credentials (2-legged) Tutorial](./oauth-client-credentials.html)
+* [Implicit (2-legged) Tutorial](./oauth-implicit.html)
+
+### Authorization Code (3-legged)
+
+* Safest one
+
+* Able to authenticate both app and user
+
+* Allows to obtain a long-lived access token  renewed with a refresh token
+
+* Access and refresh token never reach the end user (stays at the web server)
+
+### Client Credentials (2-legged)
+
+* Used when the client has been developed by the same authority as the authorization server
+
+* For scenarios in which authenticating the end user is not required.
+
+* Able to authenticate the app (no user authorization).
+
+### Implicit (2-legged)
+
+* For browser-based or mobile apps, typically running JavaScript
+
+* Able to authenticate the User only (not the app! any app could use the token => insecure)
+
+* Browser (or mobile app) receives access token, not the app server
+
+* No refresh token granted (because it's less secure)
+
+* Attention! This type of authorization should only be used if no other type of authorization is available. It is the least secure because the access token is exposed (and therefore vulnerable) on the client side.
+
+##  How do i use it?
+
+* Authorization Code: First, you need an authorization code ```c/oauth2/auth``` before you can call the get access token method. See the **[OAuth Authorization Code Tutorial](./oauth-authorisation-code.html)** and learn how to get an authorization code and how to use it.
+
+* Client Credetials: Call the access token method directly. No code is needed. It is important to use ```grant_type=client_credential```. Have look at the **[OAuth Client Credential Tutorial](./oauth-client-credentails.html)**.
```

Comment:
1) consistent naming: line 58: "call the **get** access token method", line 60: "call the access token method"

2) ```grant_type=client_credential``` -> client_credential**s**